packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const [
+    componentTree,
+    getComponentMetadata,
+    activePage,
+    UUIDToFileMetadata,
```

Comment:
where are `activePage` and `UUIDToFileMetadata` used?

Reply:
They aren't used explicitly, but I wanted to call the useEffect hook to update the style when we change the active page or FileMetadatas are updated!  But I think you have a point about the active page one.  That's not necessary since we're already watching if the page CSS is changing directly.  The filemetata data check we will still need in case users change their CSS for HMR.

Reply:
Does `getComponentMetadata` change if `UUIDToFileMetadata` changes in the store? And also, I thought we weren't handling HMR for CSS?

Reply:
Unfortunately, I found that `getComponentMetadata` doesn't change when `UUIDToFileMetadata` changes.

For HMR, we currently have 2 potential use cases: 
1) the user updates a component file by adding or removing CSS imports.  We can currently support HMR for this, unless the file is a completely new CSS file.  That is just limited by the fact that we haven't added the styletag to Studio's parent head since we only do that once on the lazy load.
2) the user changes a CSS file, which is the thing we can't support and Tom says we don't care about at the moment haha

Do you think this would be confusing and better to not support at all, or is it the case that better some than none?

Reply:
I left another comment that includes a suggestion for how we can get around watching `UUIDToFileMetadata` and adding it to the dependency array. It would be nice to have HMR support if it "comes out of the box" almost, even if it is limited. My only hesitation is that I prefer not to rely on the presence of an unused dependency in the dep array to trigger updates because it would be easy down the road to forget why it was included and remove it since it's not actually used in the callback. Another alternative could be to just add a comment explaining why it's included in the dep array

Reply:
Chatted offline, see[ this comment](https://github.com/yext/studio/pull/412#discussion_r1366103331), where we remove the need for subscribing to `UUIDToFileMetadata`